lib/ts/recipe/thirdpartypasswordless/types.ts

```diff
@@ -0,0 +1,295 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Imports.
+ */
+import { FeatureBaseConfig, NormalisedBaseConfig, Styles } from "../../types";
+import {
+    GetRedirectionURLContext as PasswordlessGetRedirectionURLContext,
+    OnHandleEventContext as PasswordlessOnHandleEventContext,
+    PreAPIHookContext as PasswordlessPreAPIHookContext,
+} from "../passwordless";
+import {
+    NormalisedConfig as PasswordlessConfig,
+    PasswordlessFeatureBaseConfig,
+    PasswordlessUser,
+    SignInUpFeatureConfigInput as PWLessSignInUpFeatureConfigInput,
+} from "../passwordless/types";
+import {
+    GetRedirectionURLContext as ThirdPartyGetRedirectionURLContext,
+    OnHandleEventContext as ThirdPartyOnHandleEventContext,
+    PreAPIHookContext as ThirdPartyPreAPIHookContext,
+} from "../thirdparty";
+import { NormalisedConfig as TPConfig, StateObject } from "../thirdparty/types";
+import Provider from "../thirdparty/providers";
+import { CustomProviderConfig } from "../thirdparty/providers/types";
+import {
+    Config as AuthRecipeModuleConfig,
+    NormalisedConfig as NormalisedAuthRecipeModuleConfig,
+    UserInput as AuthRecipeModuleUserInput,
+    User,
+    UserInputOverride as AuthRecipeUserInputOverride,
+} from "../authRecipeWithEmailVerification/types";
+import PWlessRecipe from "../passwordless/recipe";
+import TPRecipe from "../thirdparty/recipe";
+import OverrideableBuilder from "supertokens-js-override";
+
+import { ComponentOverride } from "../../components/componentOverride/componentOverride";
+import { ComponentOverrideMap as PasswordlessOverrideMap } from "../passwordless/types";
+import { ComponentOverrideMap as ThirdPartyOverrideMap } from "../thirdparty/types";
+import { Header } from "./components/themes/signInUp/header";
+import { CountryCode } from "libphonenumber-js";
+
+type WithRenamedProp<T, K extends keyof T, L extends string> = Omit<T, K> & {
+    [P in L]: T[K];
+};
+
+export type ComponentOverrideMap = Omit<PasswordlessOverrideMap, "PasswordlessSignInUpHeader"> &
+    Omit<ThirdPartyOverrideMap, "ThirdPartySignUpFooter" | "ThirdPartySignUpHeader"> & {
+        ThirdPartyPasswordlessHeader?: ComponentOverride<typeof Header>;
+    };
+
+// TODO: move the definition here if necessary
+export type SignInUpFeatureConfigInput = WithRenamedProp<
+    PWLessSignInUpFeatureConfigInput,
+    "emailOrPhoneFormStyle",
+    "providerAndEmailOrPhoneFormStyle"
+> & {
+    providers?: (Provider | CustomProviderConfig)[];
+};
+
+export type UserInput = (
+    | {
+          contactMethod: "EMAIL";
+
+          validateEmailAddress?: (email: string) => Promise<string | undefined> | string | undefined;
+
+          signInUpFeature?: SignInUpFeatureConfigInput;
```

Comment:
This is called `signInAndUpFeature` in thirdpartyemailpassword. But in `passwordless` recipe, it is called `signInUp`.. which one should we pick?

Reply:
Either is fine I think, I went with the one from passwordless because it uses more props from there.

Reply:
Okay!